### DISCUSSION TITLE
Prevent auto upgrading of packages when dependency is upgraded

### DISCUSSION THREAD

**Discussion by J-M0** [Homebrew/discussions#488] [Everyday usage]
Is there a way to prevent packages from being upgraded when a dependency is upgraded?

When running `brew` in CI, this can cause builds to run much longer than they need to if one of the packages I upgrade is a dependency of several other packages I don't care about. For instance, `openssl@1.1` is one of my dependencies and is also used by many other formulae. This means that every time my CI runs, `brew` tries to upgrade `openssl@1.1` and also upgrades any formula that depends on it, meaning that in order to test my Python project I have to wait for `curl`, `azure-cli`, `php`, etc. to get upgraded before running my tests.

**Comment by SMillerDev** [comment]
There's no way to do this, I'd recommend removing all but the required formula using `brew bundle` for example.
